Merge branch 'master' into tomoxlending

thanhnguyennguyen opened the pull request.

ngtuna commented on core/blockchain.go:
```diff
@@ -1590,3 +1593,57 @@ func (bc *BlockChain) GetClient() (*ethclient.Client, error) {
 	return bc.Client, nil
 }
+
+func (bc *BlockChain) UpdateM1() {
```
Function should return error state. In golang normally we don't do error report inside the function; instead we do it at outer level. So you should update like below:

```go
func (bc *BlockChain) UpdateM1() error {
     //do something here
     if err != nil {
            return err
     }
     return nil
}

// somewhere else calls to UpdateM1() like below
err := UpdateM1()
if err != nil {
       log.Error("error message")
}
```

ngtuna commented:
Reviewed via pair-programming. Merging now.

thanhson1085 committed:
rename USD to USDT

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhnguyennguyen committed:
tomox: update getCancelFee

thanhson1085 committed:
Merge pull request #129 from thanhnguyennguyen/update-getCancelfee

tomox: update getCancelFee

thanhson1085 committed:
fix bigint issue

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhnguyennguyen committed:
tomox: add cancelFee to relayerOwner

thanhson1085 committed:
Merge pull request #135 from thanhnguyennguyen/tomox-cancelFee-to-relayerOwner

tomox: add cancelFee to relayerOwner

thanhson1085 committed:
fix deposit function

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhnguyennguyen committed:
tomox: remove invalid orders from pool

thanhson1085 committed:
Merge pull request #138 from thanhnguyennguyen/remove-invalid-orders-from-pool

tomox: remove invalid orders from pool

anhntbk08 committed:
[Fix][TRC21] can't do transaction with full balance

thanhson1085 committed:
fix case nonTomoPairs

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhson1085 committed:
Merge pull request #140 from anhntbk08/master

[Fix][TRC21] can't do transaction with full balance

nghiatomo committed:
verify cancel order id in order pool

thanhnguyennguyen committed:
tomox: add failed order to block

Validate order in front of the pool only. When masternode grasp orders from pool to create block, we always add them to block even it is executed failed

thanhson1085 committed:
Merge pull request #146 from thanhnguyennguyen/add-fail-order-to-block

tomox: add failed order to block

thanhson1085 committed:
Update order_pool.go

thanhson1085 committed:
Merge pull request #145 from nghiatomo/master

verify cancel order id in order pool

thanhnguyennguyen committed:
tomox: Stop SDK node when failing to update orders to mongodb

thanhson1085 committed:
Merge pull request #148 from thanhnguyennguyen/stop-sdknode-if-insert-mongodb-fail

tomox: Stop SDK node when failing to update orders to mongodb

nghiatomo committed:
tomox.getOrderTxMatchByHash

thanhson1085 committed:
Merge pull request #150 from nghiatomo/master

tomox.getOrderTxMatchByHash

nguyenbatam committed:
fix debug transaction

thanhson1085 committed:
Merge pull request #155 from nguyenbatam/fix_debug_transaction

fix debug transaction

thanhson1085 committed:
default mainnet

thanhson1085 committed:
Merge branch 'master' of github.com:tomochain/tomochain

thanhnguyennguyen committed:
Merge branch 'master' into tomoxlending
